### PR TITLE
3.1 show map

### PR DESCRIPTION
**Pull request by jmollyj**
Project Walking/Source code/.idea/gradle.xml : Changes not added by me but automatically by Android Studio

Project Walking/Source code/.idea/misc.xml : Changes not added by me but automatically by Android Studio

Project Walking/Source code/app/build.gradle : I added dependecies for the osmdroid library

Project Walking/Source code/app/src/main/AndroidManifest.xml : Changes not added by me but automatically by Android Studio

Source code/app/src/main/java/com/teamapricot/projectwalking/Navigation/View/MapActivity.java : Code for the map viewer

Project Walking/Source code/app/src/main/res/layout/activity_map.xml : layout code for the map to show

### CODE REVIEW

**Review by Erik142 — CHANGES_REQUESTED** [review]
I don't have a lot to comment on, good job! The merge conflict has to be solved though. I think it would be best if you did it yourself, because practice makes perfect! 😄 But if you want, I can take care of it for you this time, just let me know. 

Other than that, please address my comments, push some new commits for your changes, and re-request my review when you feel done! 👍🏻

File: Project Walking/Source code/app/src/main/res/layout/activity_main.xml
Comment (jmollyj) [reply Project Walking/Source code/app/src/main/res/layout/activity_main.xml:23]: Im unsure where the directory MapView is in this case

**Comment by jmollyj** [comment]
I tried fixing the auto merge issue but it doesnt seem like it worked out. Ill probably need some additional help with that

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [inline Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:33]: This should be `protected` instead of `public`, since that is what the base class assumes. Otherwise it won't be overridden correctly. The base class in this case is simply the `Activity` class, see the link for details:

https://developer.android.com/reference/android/app/Activity.html#protected-methods

And here is the corresponding class inheritance:

https://developer.android.com/reference/androidx/appcompat/app/AppCompatActivity

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [inline Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:55]: This should be `protected` instead of `public`, since that is what the base class assumes. Otherwise it won't be overridden correctly. The base class in this case is simply the `Activity` class, see the link for details:

https://developer.android.com/reference/android/app/Activity.html#protected-methods

And here is the corresponding class inheritance:

https://developer.android.com/reference/androidx/appcompat/app/AppCompatActivity

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [inline Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:65]: This should be `protected` instead of `public`, since that is what the base class assumes. Otherwise it won't be overridden correctly. The base class in this case is simply the `Activity` class, see the link for details:

https://developer.android.com/reference/android/app/Activity.html#protected-methods

And here is the corresponding class inheritance:

https://developer.android.com/reference/androidx/appcompat/app/AppCompatActivity

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [inline Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:61]: See the comment about the `MapController` class, but this should be moved to `MapController.onResume`.

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [inline Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:71]: See the comment about the `MapController` class, but this should be moved to `MapController.onPause`.

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [inline Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:86]: This code does not belong here! Did it move here when you merged the code? This should be inside `MainActivity.onCreate`, otherwise I don't think that the code is going to compile, and it will definitely not work as intended! 😅

File: Project Walking/Source code/app/src/main/res/layout/activity_main.xml
Comment (Erik142) [inline Project Walking/Source code/app/src/main/res/layout/activity_main.xml:24]: The element `<com.google.android.material.floatingactionbutton.FloatingActionButton ... />` is missing here, so right now there is no button for the camera in your branch. The button exists in the master branch, and the rest of the camera code is in your branch, so the button must have disappeared when you merged the code. I suggest that you revert your last commit, start over with the changes from the previous review, and add them one by one. 

One commit for fixing activity_main.xml. 
One commit for adding the `MapController` class. 
One commit for using the `MapController` class in `MainActivity`. 
One commit for merging the master branch with your branch. 

I don't want to come across as unpleasant or rude in any way, but if the commits were done in that way before, you would only have to revert the "merge commit", but you could have kept your other changes. Now, since everything was made in one commit, you will have to start over and take care of the comments from the first review again, plus this review 😅

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [inline Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:28]: Move this to the `MapController` class.

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [inline Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:22]: When you move the map code to the `MapController` class, some of these imports will become unnecessary. If they're "greyed out", remove them 👍🏻

**Review by Erik142 — CHANGES_REQUESTED** [review]
Good progress! I believe that something strange happened when you merged the master branch with this branch locally, right? Some things seem to have disappeared, and some things seem to have moved out of their place. So I'm sorry to say, but I think you have to revert your latest commit, and do this one more time. I think the last part of my comment regarding "activity_main.xml" summarizes this:

One commit for fixing activity_main.xml. 
~One commit for adding the `MapController` class.~
~One commit for using the `MapController` class in `MainActivity`.~
One commit for merging the master branch with your branch. 

I don't want to come across as unpleasant or rude in any way, but if the commits were done in that way before, you would only have to revert the "merge commit", but you could have kept your other changes. Now, since everything was made in one commit, you will have to start over and take care of the comments from the first review again, plus this review 😅

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [reply Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:47]: Will be handled in #40.

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [reply Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:51]: Will be handled in #40.

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [reply Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:71]: Will be handled in #40.

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [reply Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:28]: Will be handled in #40.

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [reply Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:22]: Will be handled in #40.

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/Navigation/MapController.java
Comment (Erik142) [reply Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/Navigation/MapController.java:10]: Will be handled in #40.

File: Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java
Comment (Erik142) [reply Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java:61]: Will be handled in #40.

**Review by Erik142 — APPROVED** [review]
LGTM!

Good job with the implementation, and GREAT job with finishing the merge conflict! 👍🏻

**Comment by Erik142** [comment]
PO @danbrn: Review approved, waiting for the final approval and merge from you.

**Pull request merged**
